### PR TITLE
Organize the cli-crashcourse project

### PR DESCRIPTION
**Pull request by rgreinho**
* Add the documentation
  Moves the tutorial inside of this repository
* Add the IP command

And reorganize our thoughts.

### CODE REVIEW

File: cmd/ip.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			log.Fatalln("Error retrieving your IP:" + err.Error() + ".")
+		}
+		fmt.Println(ip)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(ipCmd)
+
+	// Here you will define your flags and configuration settings.
```
Comment (laconc) [inline cmd/ip.go]: Should we get rid of all these stock comments?

File: cmd/ip.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ip, err := api.Call()
+		if err != nil {
+			log.Fatalln("Error retrieving your IP:" + err.Error() + ".")
```
Comment (laconc) [inline cmd/ip.go]: How about if we add a space after the colon for formatting purposes?

File: pkg/api.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package api
+
+import (
+	"io/ioutil"
+	"net/http"
+)
+
+const baseURL = "https://api.ipify.org?format=json"
```
Comment (laconc) [inline pkg/api.go:8]: I do love a baseURL variable when making API calls, and I think that's a good best practice, but considering there is a single endpoint here, perhaps we should just have it inside the function in a variable called url.

**Review by laconc — APPROVED** [review]
We had a very productive session the other day! I made a few comments here but none are really breaking in any way, so approving.

**Comment by laconc** [comment]
I did also notice some spelling and grammar issues in the doc but we can polish all that up in the end.

File: cmd/ip.go (hunk 2 above)
Comment (rgreinho) [reply cmd/ip.go]: Addressed.

File: cmd/ip.go (hunk 1 above)
Comment (rgreinho) [reply cmd/ip.go]: Addressed.

File: pkg/api.go (hunk 3 above)
Comment (rgreinho) [reply pkg/api.go:8]: Although, you are totally right, Let's stick with the baseURL approach for now. We'll do a code clean up once the tutorial gets closer to completion.

**Comment by rgreinho** [comment]
Awesome! We should fix them when we find them, but we'll proof-read everything once we're satisfied with the content.

**Pull request merged**